python/ray/train/tests/test_huggingface_predictor.py

```diff
@@ -62,6 +63,52 @@ def test_repr(tmpdir):
     assert pattern.match(representation)
 
 
+def test_huggingface_checkpoint(tmpdir, ray_start_runtime_env):
+    model_config = AutoConfig.from_pretrained(model_checkpoint)
+    model = AutoModelForCausalLM.from_config(model_config)
+    tokenizer = AutoTokenizer.from_pretrained(tokenizer_checkpoint)
+    preprocessor = DummyPreprocessor()
+    preprocessor.attr = 1
+
+    checkpoint = HuggingFaceCheckpoint.from_model(
+        model, tokenizer, path=tmpdir, preprocessor=preprocessor
+    )
+    checkpoint_model = checkpoint.get_model(AutoModelForCausalLM)
+    checkpoint_tokenizer = checkpoint.get_tokenizer(AutoTokenizer)
+    checkpoint_preprocessor = checkpoint.get_preprocessor()
+
+    @ray.remote
+    def test(model, tokenizer):
+        os.chdir(tmpdir)
+        # Ensure that model outputs are deterministic
+        model_config = AutoConfig.from_pretrained(model_checkpoint)
+        model = AutoModelForCausalLM.from_config(model_config)
+        model.eval()
```

Comment:
we shouldn't need to do this - it should happen inside the HF pipeline itself.

Reply:
Removed. Also fixed an error where the model shouldn't have been loaded again within this test function.